sdk/python/feast/offline_server.py

```diff
@@ -126,67 +131,165 @@ def do_get(self, context, ticket):
-            name_aliases = command["name_aliases"]
-            logger.debug(f"do_get: name_aliases is {name_aliases}")
-            feature_refs = command["feature_refs"]
-            logger.debug(f"do_get: feature_refs is {feature_refs}")
-            project = command["project"]
-            logger.debug(f"do_get: project is {project}")
-            full_feature_names = command["full_feature_names"]
-            feature_views = self.list_feature_views_by_name(
-                feature_view_names=feature_view_names,
-                name_aliases=name_aliases,
-                project=project,
-            )
-            logger.debug(f"do_get: feature_views is {feature_views}")
+        try:
+            if api == OfflineServer.get_historical_features.__name__:
+                df = self.get_historical_features(command, key).to_df()
+            elif api == OfflineServer.pull_all_from_table_or_query.__name__:
+                df = self.pull_all_from_table_or_query(command).to_df()
+            elif api == OfflineServer.pull_latest_from_table_or_query.__name__:
+                df = self.pull_latest_from_table_or_query(command).to_df()
+            else:
+                raise NotImplementedError
+        except Exception as e:
+            logger.exception(e)
+            traceback.print_exc()
+            raise e
 
-            logger.info(
-                f"get_historical_features for: entity_df from {entity_df.index[0]} to {entity_df.index[len(entity_df)-1]}, "
-                f"feature_views is {[(fv.name, fv.entities) for fv in feature_views]}"
-                f"feature_refs is {feature_refs}"
-            )
+        table = pa.Table.from_pandas(df)
 
-            try:
-                training_df = (
-                    self.store._get_provider()
-                    .get_historical_features(
-                        config=self.store.config,
-                        feature_views=feature_views,
-                        feature_refs=feature_refs,
-                        entity_df=entity_df,
-                        registry=self.store._registry,
-                        project=project,
-                        full_feature_names=full_feature_names,
-                    )
-                    .to_df()
-                )
-                logger.debug(f"Len of training_df is {len(training_df)}")
-                table = pa.Table.from_pandas(training_df)
-            except Exception as e:
-                logger.exception(e)
-                traceback.print_exc()
-                raise e
+        # Get service is consumed, so we clear the corresponding flight and data
+        del self.flights[key]
+        return fl.RecordBatchStream(table)
 
-            # Get service is consumed, so we clear the corresponding flight and data
-            del self.flights[key]
+    def offline_write_batch(self, command, key):
+        feature_view_names = command["feature_view_names"]
+        assert len(feature_view_names) == 1
```

Comment:
thanks for using assert.
you can even add a message set in the assert exception when the condition is false:

assert x != "hello", "x should be 'hello'"

Reply:
I know will add